src/Propagation/ServiceName/src/ServiceNamePropagator.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\Propagation\ServiceName;
+
+use OpenTelemetry\Context\Context;
+use OpenTelemetry\Context\ContextInterface;
+use OpenTelemetry\Context\Propagation\ArrayAccessGetterSetter;
+use OpenTelemetry\Context\Propagation\PropagationGetterInterface;
+use OpenTelemetry\Context\Propagation\PropagationSetterInterface;
+use OpenTelemetry\Context\Propagation\TextMapPropagatorInterface;
+use OpenTelemetry\SDK\Resource\Detectors;
+use OpenTelemetry\SemConv\ResourceAttributes;
+
+/**
+ * Provides a TextMapPropagator implementation for service.name
+ *
+ * @see https://opentelemetry.io/docs/specs/semconv/resource/#service
+ */
+final class ServiceNamePropagator implements TextMapPropagatorInterface
+{
+    private const SERVICE_NAME = 'service.name';
+
+    private const FIELDS = [
+        self::SERVICE_NAME,
+    ];
+
+    private static ?self $instance = null;
+
+    public static function getInstance(): self
+    {
+        if (null === self::$instance) {
+            self::$instance = new self();
+        }
+
+        return self::$instance;
+    }
+
+    /**
+     * @suppress PhanUndeclaredClassAttribute
+     */
+    #[\Override]
+    public function fields(): array
+    {
+        return self::FIELDS;
+    }
+
+    /**
+     * @suppress PhanUndeclaredClassAttribute
+     */
+    #[\Override]
+    public function inject(&$carrier, ?PropagationSetterInterface $setter = null, ?ContextInterface $context = null): void
+    {
+        $setter ??= ArrayAccessGetterSetter::getInstance();
+
+        $detector = new Detectors\Service();
```

Comment:
This will need some thought. Resources including `service.name` can come from a variety of sources (for example, a yaml file in declarative config). I think the best source is the resource that's associated with the tracer, but that's not currently accessible. Perhaps service name needs to be added to SpanContext, so that it can be accessed in the same way that other components are for context propagation? That's a spec change, we won't just add it to our API and SDK.

Reply:
Yes, definitely the best source is from `Tracer`. Agreed it requires a spec change and involves a change in the API and SDK. I will monitor the discussion of the semconv to see what is the recommendation to get the service name.

Reply:
Yes, this is the semconv [PR](https://github.com/open-telemetry/semantic-conventions/pull/2495). I saw you put comments there.

Reply:
The most recent comments on that PR suggest that it should be [provided by the user](https://github.com/open-telemetry/semantic-conventions/pull/2495#discussion_r2209133620) and it may be the service name or some completely different user-defined value. We can wait for that to be agreed upon, but to me that means it's not coming from a detector, but actual user input. For an auto-instrumentation, that could be an env var or from declarative configuration.

Reply:
Definitely. We can wait for the agreement. If it is a set of user-defined values, we can put any key value pair in the comment.

While we are waiting for the agreement, I wonder if I can do anything to move this forward. 

How about I remove all service name propagator reference and we just prepend / append trace context to the query statement (except `prepare statement` scenario, where we need to put low cardinality comment), and I am happy to do the same for Instrumentation/MySqli? 

Does it sound good to you?

Reply:
Yes, let's remove the service name propagator, since it seems the proposed solution is not as sophisticated as that. I don't think you should start on mysqli until upstream (spec/semconv) PRs are finalised. I'd rather see one working prototype that can test out the proposed changes thoroughly, and be implemented well.
What about working on how to opt-in/out and how to do different types of propagation for different DBMS (eg sql-server)?

Reply:
Thanks! I removed the service name propagator and all references to it.

Now I have `OTEL_PHP_INSTRUMENTATION_PDO_SQL_COMMENTER_DATABASE` to select which database(s) opt-in to sqlcommenter feature. 

However, as both prepend/append sql comments and set_context_info ways are also working for sql-server and `set_context_info` is specific to sql-server, I am now thinking to add another configuration directive to opt-in `set_context_info` for sql-server (By default using prepend/append sql comments). It seems that I am adding more and more configuration directives and not sure if it is a good idea to do so.

What do you think?

Reply:
I think the recommendation for MS SQL Server is to use `CONTEXT_INFO` instead of sqlcomment as propagation method. As mentioned in https://github.com/open-telemetry/opentelemetry-php-contrib/pull/406#issuecomment-3081320457 i really think we should limit this first cut to just the DBMS we're testing against, e.g. MySQL and maybe Postgres.

Reply:
Done. Limited to `postgresql` and `mysql` only

Reply:
@brettmc The spec mentioned the `The instrumentation implementation SHOULD append the comment to the end of the query`. Should I remove the option to allow prepending the query?

Reply:
It goes on to say that `Semantic conventions for individual database systems MAY specify different format, which may include different position, encoding, or schema, depending on the specific database system's requirements or preferences.`
Do we know of any databases where prepending could cause an issue, or is it just user preference (in which case, removing the option seems ok to me) ?